winchan.js

```diff
@@ -116,8 +116,8 @@ var WinChan = (function() {
         var iframe;
 
         // sanity check, are url and relay_url the same origin?
-        var origin = extractOrigin(opts.url);
-        if (origin !== extractOrigin(opts.relay_url)) {
+        var origin = opts.origin || extractOrigin(opts.url);
+        if (!opts.origin && origin !== extractOrigin(opts.relay_url)) {
```

Comment:
I'm not sure about this change. The error returned in the callback say that the url and relay_url origins must match. Otherwise I'm not sure that postMessage actually works.